src/components/Vote/index.jsx

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import connect from 'utils/connect';
+import { voteOnArticle, initArticleVoteCount } from 'modules/vote';
+import Icon from 'components/Icon';
+
+import './Vote.scss';
+
+/**
+ * Component for article vote
+ * @summary composes the upvote and the downvote icons along with the vote count
+ */
+class Vote extends React.Component {
+  /**
+    * @name Vote propTypes
+    * @type {propTypes}
+    *
+    * @param {Object} props - React PropTypes
+    *
+    * @property {string} articleId -id of the article to vote on
+    * @property {string} activeVoteType - article voteType by the current user
+    * @property {Function} voteOnArticle -action creator to vote on an article
+    * @property {Function} articleVote -redux state map to props
+    *
+    * @return {Array} React propTypes
+    */
+  static propTypes = {
+    articleId: PropTypes.string.isRequired,
+    activeVoteType: PropTypes.string,
+    voteOnArticle: PropTypes.func.isRequired,
+    initArticleVoteCount: PropTypes.func.isRequired,
+    articleVote: PropTypes.isRequired,
+    upVoteCount: PropTypes.number,
+    downVoteCount: PropTypes.number
+  }
+
+  static defaultProps = {
+    activeVoteType: null,
+    upVoteCount: 0,
+    downVoteCount: 0
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = { activeVoteType: props.activeVoteType };
+  }
+
+  componentDidMount() {
+    const { initArticleVoteCount: articleVoteCount, upVoteCount, downVoteCount } = this.props;
+
+    articleVoteCount({ upVoteCount, downVoteCount });
+  }
+
+  /**
+    * handles vote action
+    * @method
+    *
+    * @param {Object} payload
+    * @param {string} payload.voteType - type of vote
+    *
+    * @return {void}
+    */
+  onVoteHandler = ({ voteType }) => {
+    const { voteOnArticle: voteOnArticleHandler, articleId } = this.props;
+    const { activeVoteType } = this.state;
+
+    const vote = voteType === activeVoteType ? 'nullVote' : voteType;
+
+    voteOnArticleHandler({ articleId, voteType: vote, prevVote: activeVoteType });
+
+    this.setState((prevState) => ({ ...prevState, activeVoteType: vote }));
+  };
+
+  render() {
+    const { articleVote: { upVoteCount, downVoteCount } } = this.props;
+    const { activeVoteType } = this.state;
+    const upVoteActive = activeVoteType === 'upVote';
+
+    return (
+      <div className="vote">
+        <div className="vote__wrapper">
+          <Icon icon="upvote" voteType="upVote" clickHandler={this.onVoteHandler} active={upVoteActive} />
+          <span className={`${upVoteActive && 'active'}`}>{upVoteCount}</span>
+        </div>
+        <div className="vote__wrapper">
+          <Icon icon="downvote" voteType="downVote" clickHandler={this.onVoteHandler} active={activeVoteType !== 'nullVote' && !upVoteActive} />
+          <span className={`${activeVoteType !== 'nullVote' && !upVoteActive && 'active'}`}>{downVoteCount}</span>
```

Comment:
Let's create a class name formatter similar to the `classnames` package for these sort of use cases